subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/impl/EmptyCurrentFileCollectionFingerprint.java

```diff
@@ -42,7 +42,7 @@ public EmptyCurrentFileCollectionFingerprint(String identifier) {
     @Override
     public boolean visitChangesSince(FileCollectionFingerprint oldFingerprint, final String title, boolean includeAdded, ChangeVisitor visitor) {
         for (Map.Entry<String, FileSystemLocationFingerprint> entry : oldFingerprint.getFingerprints().entrySet()) {
-            if (!visitor.visitChange(DefaultFileChange.removed(entry.getKey(), title, entry.getValue().getType()))) {
+            if (!visitor.visitChange(DefaultFileChange.removed(entry.getKey(), title, entry.getValue().getType(), entry.getValue().getNormalizedPath()))) {
```

Comment:
I think it would make sense to move the visiting out of the `if` condition now.

Reply:
Done.